server/controllers/user.js

```diff
@@ -0,0 +1,165 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+const secretKey = process.env.SECRET_TOKEN_KEY;
+
+const saltRounds = 10;
+
+const user = require('../models').User;
+const document = require('../models').Document;
+
+const findWithDocuments = (method, params) => {
+  const includeDocuments = {
+    include: [{
+      model: document,
+      as: 'documents',
+    }],
+  };
+
+  if (params) {
+    return user[method](params, includeDocuments);
+  }
+
+  return user[method](includeDocuments);
+}
+
+module.exports = {
+
+  //create a new user
+
+  create(req, res) {
+    let firstName = req.body.firstName;
+    let lastName = req.body.lastName;
+    let userName = req.body.userName;
+    let email = req.body.email;
+    let password = bcrypt.hashSync(req.body.password, saltRounds);
+
+    if (!firstName || !lastName || !userName || !email || !password) {
+      return res.status(400).json({ message: 'Enter All Required Fields' });
+    } else if (!(/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(req.body.email))) {
+      return res.status(401).json({ message: 'Please Enter A Valid Email Address' });
```

Comment:
**Nitpick** - The status code here should probably also be `400` since and invalid email is client error.